module/os/linux/zfs/zpl_file.c

```diff
@@ -811,26 +811,27 @@ zpl_fadvise(struct file *filp, loff_t offset, loff_t len, int advice)
 	return (error);
 }
 
-#define	ZFS_FL_USER_VISIBLE	(FS_FL_USER_VISIBLE | ZFS_PROJINHERIT_FL)
-#define	ZFS_FL_USER_MODIFIABLE	(FS_FL_USER_MODIFIABLE | ZFS_PROJINHERIT_FL)
+#define	ZFS_FL_USER_VISIBLE	(FS_FL_USER_VISIBLE | FS_PROJINHERIT_FL)
+#define	ZFS_FL_USER_MODIFIABLE	(FS_FL_USER_MODIFIABLE | FS_PROJINHERIT_FL)
 
 static uint32_t
-__zpl_ioctl_getflags(struct inode *ip)
+__zpl_ioctl_getflags(struct inode *ip, boolean_t is_xflags)
```

Comment:
Let's create a separate `__zpl_ioctl_getxflags()` version of this function, the same as you did for `__zpl_ioctl_setxflags()`.  This way we can return the correct `zfsxattr_t` type, and it makes it clear that *flags and *xflags are different things.